tls.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2017-present Kirill Danshin and Gramework contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+
+package gramework
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"math/big"
+	"time"
+)
+
+const localhost = "localhost"
+
+func selfSignedCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	if clientHello.ServerName != localhost {
+		return nil, fmt.Errorf("self-signed certificate for %q not permitted", clientHello.ServerName)
+	}
+
+	priv, err := rsa.GenerateKey(rand.Reader, 1024)
```

Comment:
how about something like this?
```suggestion
	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```